### ISSUE TITLE
Incorrect nullability warnings reported for compound assignments with lifted struct operators

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#58201]
**Version Used**: 
```
Branch main (4 Dec 2021)
Latest commit 04dda9f by Jonathon Marolf:
Merge pull request #58110 from jmarolf/infrastructure/verify-ready-for-pr-scripts
```

**Steps to Reproduce**:

Compile the following code using .net6
```cs
#nullable enable
using System.Numerics;

Vector2? v = GetVector();
if (v != null)
  v += new Vector2(1, 1); // false CS8620

Vector2? GetVector() => null;
```

**Expected Behavior**:
No warnings.
1. `v` is not null because it's explicitly checked just above the compound assignment, so there's no reason to emit the warning.
2. even if `v` is null the used operator is a lifted operator that should work with `Nullable<Vector2>` without any issues. In fact, Roslyn recognizes that in regular additions and doesn't report nullable variables used as addition operands there but somehow fails to apply the same logic for compound assignments. I.e. the following code correctly has no warnings:
```cs
#nullable enable
using System.Numerics;

Vector2? v = null;
v = v + new Vector2(1, 1); // no warnings
```
3. Interestingly, the warning complains about "differences in the nullability of reference types" but there are simply no reference types in the `Nullable<Vector2>` type at all. It seems that even the warning message indicates that it shouldn't even be applicable to such cases at all.

**Actual Behavior**:
`warning CS8620: Argument of type 'Vector2?' cannot be used for parameter 'left' of type 'Vector2' in 'Vector2 Vector2.operator +(Vector2 left, Vector2 right)' due to differences in the nullability of reference types.` is reported for `v` in `v += new Vector2(1, 1);`
